### DISCUSSION TITLE
useMutation's `isPending` stays on `undefined` after calling mutate

### DISCUSSION THREAD

**Discussion by Slmii** [TanStack/query#6959] [Q&A]
I have upgraded from v4 to v5 and having trouble with getting the correct values from a `useMutation`. I have a simple mutation in my app
```js
const { mutate, isPending, status } = useMutation({
		mutationFn: () => {
			return new Promise(resolve => {
				setTimeout(() => {
					resolve(10);
				}, 3000);
			});
		}
	});
```
when I run `mutate` the `isPending` stays on `undefined` and `status` becomes `loading` instead of `pending`. Strange thing is that when I run this on codesandbox it works as expected...

Is there something I am forgetting with an upgrade? I really cant figure this out.

**Comment by TkDodo** [comment]
Can you show a running reproduction?

**Comment by Slmii** [comment]
@TkDodo so I cannot reproduce this in a sandbox, but here's a video of it

https://github.com/TanStack/query/assets/9927849/d31efdd6-5cd1-4495-b760-e6d6c135f73f

Here's the code thats running, on version 5.22.2
```js
const Test = () => {
	const { mutate, isPending, status } = useMutation({
		mutationFn: () => {
			return new Promise(resolve => {
				setTimeout(() => {
					resolve(10);
				}, 3000);
			});
		}
	});

	console.log({ isPending, status });

	return (
		<>
			<button onClick={() => mutate()}>TEST</button>
			<p>Typeof Pending: {typeof isPending}</p>
			<p>Boolean Pending: {isPending ? 'true' : 'false'}</p>
			<p>Status: {status}</p>
		</>
	);
};
```

Result in codesandbox, exact same code
<img width="220" alt="image" src="https://github.com/TanStack/query/assets/9927849/31c22c48-fafc-45b8-a79b-adc124137c3a">

**Reply by Slmii** [reply]
oh wow, it works now after a server restart 🙃

**Comment by osmancakir** [comment]
I also have the same issue now. isPending is undefined but isLoading works strangely fine.  unfortunately restarting the server didn't help me


```javascript
// login hook
const login = async ({username, password}) => {
  const {code_verifier, code_challenge} = await pkceChallenge()

  const dataToPost = {
    username,
    password,
    response_type: 'code',
    grant_type: 'authorization_code',
    client_id: CLIENT_ID,
    scope: 'openid profile',
    code_challenge,
    code_challenge_method: 'S256',
  }

  const res = await axios.post('/api/v1/auth/login', dataToPost, {
    maxRedirects: 0,
  })
  let code = new URLSearchParams(res.request?.responseURL.split('?')[1]).get('code')

  let token_data = {
    grant_type: 'authorization_code',
    scope: 'openid profile',
    code,
    client_id: CLIENT_ID,
    code_verifier,
  }
  const {data: tokenData} = await axios.post('/api/v1/oauth/token', token_data)
  setTokenData(tokenData)

  const {data: userData} = await axios.get('/api/v1/auth/me')
  const {data: preferencesData} = await axios.get(`/api/v1/preferences`)

  return {userData, preferencesData}
}

export const useLogin = () =>
  useMutation({
    mutationFn: login,
    onSuccess: ({userData, preferencesData}) => {
      store.dispatch(fetchMeSuccess(userData))
      setLanguagePreference(userData.preferred_language)
      setTimezonePreference(userData.preferred_timezone)
      const {table_configs, id} = preferencesData
      store.dispatch(setInitialTablePreferences({id, table_configs}))
      store.dispatch(loginSuccess())
      store.dispatch(updateLocation(getPrevLocation() || '/'))
    },
    onError: (error) => {
      store.dispatch(loginError(error))
    },
  })
```

and I call the hook in my FormLogin as ` const {mutate: login, isPending, error} = useLogin()`
Versions: 
   - "@tanstack/react-query": "^5.24.1",
   -  "@tanstack/react-query-devtools": "^5.24.1",
When I console.log "status" it is `idle` -> `loading` -> `success`.

**Reply by TkDodo** [reply]
if you have an issue, please show a codesandbox reproduction.  There is no `loading` state for mutations in v5:

v5:
https://github.com/TanStack/query/blob/61f4d747883ce67cf4edcc65f9a7693fa2bc7b6a/packages/query-core/src/types.ts#L692

v4:
https://github.com/TanStack/query/blob/716c5d8b096c624f6643dc9f263fbdd6f0565021/packages/query-core/src/types.ts#L555
